src/services/DfusionService.ts

```diff
@@ -0,0 +1,234 @@
+import Web3 from 'web3'
+
+import Logger from 'helpers/Logger'
+import { ContractEventLog } from 'contracts/types'
+import { OrderPlacement, StablecoinConverter } from 'contracts/StablecoinConverter'
+import packageJson from '../../package.json'
+import { BigNumber } from 'bignumber.js'
+import { Erc20Contract } from 'contracts/Erc20Contract'
+
+// TODO: Create common lib with the API/repo
+import tokenList = require('@gnosis.pm/dex-react/src/api/tokenList/tokenList.json')
+
+const log = new Logger('service:dfusion')
+
+export interface Params {
+  stableCoinConverterContract: StablecoinConverter
+  erc20Contract: Erc20Contract
+  web3: Web3
+}
+
+export interface DfusionService {
+  // Watch events
+  watchOrderPlacement(params: WatchOrderPlacementParams): void
+
+  // Basic info
+  getAbout(): Promise<AboutDto>
+}
+
+export interface WatchOrderPlacementParams {
+  onNewOrder: (order: OrderDto) => void
+  onError: (error: Error) => void
+}
+
+interface TokenDto {
+  name?: string
+  symbol?: string
+  decimals: number
+  address: string
+}
+
+interface AboutDto {
+  blockNumber: number
+  networkId: number
+  nodeInfo: string
+  version: string
+  stablecoinConverterAddress: string
+}
+
+export interface OrderDto {
+  owner: string
+  buyToken: TokenDto
+  sellToken: TokenDto
+  validFrom: Date
+  validUntil: Date
+  validFromBatchId: BigNumber
+  validUntilBatchId: BigNumber
+  priceNumerator: BigNumber
+  priceDenominator: BigNumber
+  event: ContractEventLog<OrderPlacement>
+}
+
+export class DfusionRepoImpl implements DfusionService {
+  private _web3: Web3
+  private _contract: StablecoinConverter
+  private _erc20Contract: Erc20Contract
+  private _networkId: number
+  private _batchTime: BigNumber
+  private _tokenCache: { [tokenAddress: string]: TokenDto } = {}
+
+  constructor (params: Params) {
```

Comment:
curious as to why you don't deconstruct in the constructor args

Reply:
In this case, I could have done it, but I prefer a simpler definition.
I like the function definition to be as simple as possible, so if I'm defining it I prefer not to extract already. I only extract in the definition if the parameters are few.

That's also way I defined `Params` in it's own place, not right there in the constructor